src/ILLink.Tasks/build/Microsoft.NET.ILLink.targets

```diff
@@ -59,38 +59,8 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <!-- Suppress warnings produced by the linker or by the ILLink Roslyn analyzer. -->
   <PropertyGroup Condition="'$(SuppressTrimAnalysisWarnings)' == 'true'">
-    <!-- RequiresUnreferenceCodeAttribute method called -->
-    <NoWarn>$(NoWarn);IL2026;IL2116</NoWarn>
-    <!-- Invalid use of DynamicallyAccessedMembersAttribute -->
-    <NoWarn>$(NoWarn);IL2041;IL2042;IL2043;IL2056</NoWarn>
-    <!-- Reference to removed attribute type -->
-    <NoWarn>$(NoWarn);IL2045</NoWarn>
-    <!-- RequiresUnreferencedCodeAttribute mismatch on virtual override -->
-    <NoWarn>$(NoWarn);IL2046</NoWarn>
-    <!-- COM marshalling warning -->
-    <NoWarn>$(NoWarn);IL2050</NoWarn>
-    <!-- Reflection intrinsics with unknown arguments -->
-    <NoWarn>$(NoWarn);IL2032;IL2055;IL2057;IL2058;IL2059;IL2060;IL2061;IL2096</NoWarn>
-    <!-- Unknown values passed to locations with DynamicallyAccessedMemberTypes -->
-    <NoWarn>$(NoWarn);IL2062;IL2063;IL2064;IL2065;IL2066</NoWarn>
-    <!-- Unsatisfied DynamicallyAccessedMembers requirements -->
-    <NoWarn>$(NoWarn);IL2067;IL2068;IL2069;IL2070;IL2071;IL2072;IL2073;IL2074;IL2075;IL2076;IL2077;IL2078;IL2079;IL2080;IL2081;IL2082;IL2083;IL2084;IL2085;IL2086;IL2087;IL2088;IL2089;IL2090;IL2091</NoWarn>
-    <!-- DynamicallyAccessedMembersAttribute mismatch on virtual override -->
-    <NoWarn>$(NoWarn);IL2092;IL2093;IL2094;IL2095</NoWarn>
-    <!-- DynamicallyAccessedMembersAttribute used on unsupported member -->
-    <NoWarn>$(NoWarn);IL2097;IL2098;IL2099;IL2106</NoWarn>
-    <!-- Unknown value passed to Expression.Property -->
-    <NoWarn>$(NoWarn);IL2103</NoWarn>
-    <!-- Multiple methods associated with state machine type or user method -->
-    <NoWarn>$(NoWarn);IL2107;IL2117</NoWarn>
-    <!-- Unannotated type derived from base type with RequiresUnreferencedCode -->
-    <NoWarn>$(NoWarn);IL2109</NoWarn>
-    <!-- Reflection access to members with DynamicallyAccessedMembers requirements -->
-    <NoWarn>$(NoWarn);IL2110;IL2111;IL2114;IL2115</NoWarn>
-    <!-- Reflection access to members with RequiresUnreferencedCode -->
-    <NoWarn>$(NoWarn);IL2112;IL2113</NoWarn>
-    <!-- Reflection access to compiler-generated code -->
-    <NoWarn>$(NoWarn);IL2118;IL2119;IL2120</NoWarn>
+    <ILLinkWarningLevel Condition="'$(ILLinkWarningLevel)' == ''">0</ILLinkWarningLevel>
```

Comment:
Maybe warn or error instead of not setting it? I think these settings are in conflict and we should probably signal there's something wrong.

Also, I think SuppressTrimAnalysisWarnings should take precedence.

Reply:
This might be different from EnableTrimAnalyzer, as I could see wanting to just disable the linker.

Reply:
Hmm. That makes sense. Maybe warn for both?

Reply:
Do we have some mechanism where `WarningLevel` influences `ILLinkWarningLevel`, or are those completely separate?
(Honestly I thought we didn't have `ILLinkWarningLevel` at all - since we only have `NoWarn` and there's no `ILLinkNoWarn`).

Reply:
> What should be the preferred way to enable analyzer warnings but not linker warnings?

I don't think this needs to be "easy" - should be rare to want this
But I guess the `SuppressTrimAnalysisWarnings`/`EnableTrimAnalyzer` makes more sense in this case.

I think I came around to what the code does in this PR as-is. It's not perfect, but at least it makes logical sense :-)